OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -572,14 +556,10 @@ private void dismissCurrentMessage(@Nullable OSInAppMessage message) {
         }
     }
 
-    private void persistInAppMessageForRedisplay(final OSInAppMessage message) {
-        //If the IAM doesn't have the re display configuration then no need to save it
-        if (!message.getDisplayStats().isRedisplayEnabled())
-            return;
-
+    private void persistInAppMessage(final OSInAppMessage message) {
         long displayTimeSeconds = System.currentTimeMillis() / 1000;
-        message.getDisplayStats().setLastDisplayTime(displayTimeSeconds);
-        message.getDisplayStats().incrementDisplayQuantity();
+        message.getRedisplayStats().setLastDisplayTime(displayTimeSeconds);
```

Comment:
Sry I think you already explained me this, but why is always saving redisplay now?

Reply:
We should treat all IAMs the same way as it creates consistency 

Even though we arent visually seeing an IAM with a limit of 1, we still need to cache it and keep track of it. Its confusing saving only specific set of IAMs to SQL when you can easily do them all.

Specifically why I did this was because I needed to know when to clean Shared Preferences after 6 months and if we arent saving all IAMs we only end up the redisplay ones